### ISSUE TITLE
Hyperparameter search and test-set

### ISSUE THREAD

**Issue by sdimi** [maxpumperla/hyperas#193]
A more high-level issue. In your examples it seems that you evaluate with the *test-set* in every iteration of the hyperparam-search.

```python
score, acc = model.evaluate(x_test, y_test, verbose=0)
```

A more fair evaluation would be to consider the *validation-test* only, since by doing that you overfit to the test-set. The test-set should be seen only once in the end, when you compare it with your baselines.

I'd suggest something along these lines:
```python
result = model.fit(X_train....)
validation_loss = np.amin(result.history['val_loss']) 
return {'loss': validation_loss, 'status': STATUS_OK, 'model': model}
```
in order to take the minimum validation loss and optimize based on that.

Am I missing something obvious?

**Comment by maxpumperla** [comment]
@sdimi that's correct, I'm just showing how the API works in the examples and am not commenting on proper methodology. But I can see how people would take my examples too literally. 

Would you mind sending a PR with an example of your choosing?

**Comment by sdimi** [comment]
Alright, just sent a PR.

**Comment by maxpumperla** [comment]
@sdimi closing this one (you "solved" it yourself) :+1:
